### PR TITLE
Upgrade OpenSSL_3 submodule to 3.1

### PR DESCRIPTION
**Pull request by dcarpente**
This contains a couple changes:
1. Updating OpenSSL_3 submodule and files produced by update.make
2. Updates to 3rdparty/openssl/CMakeLists.txt to include new assembly scripts, and a change to avoid duplicating a list of assembly scripts twice
3. Update to include a couple more tests that were added with OpenSSL 3.1
4. Acting on these two comments from Anand I missed in a previous PR:
https://github.com/openenclave/openenclave/pull/4778#pullrequestreview-1495907663

### CODE REVIEW

**Comment by dcarpente** [comment]
bors try

File: tests/openssl/tests_3.unsupported [hunk 1]
```diff
@@ -7,8 +7,6 @@
 ../../3rdparty/openssl/openssl/test/asn1_time_test.c
 # The MS CryptoAPI engine (capieng) is disabled.
 ../../3rdparty/openssl/openssl/test/cmsapitest.c
-# PSK is disabled by OE.
-../../3rdparty/openssl/openssl/test/dtls_mtu_test.c
```
Comment (mingweishih) [inline tests/openssl/tests_3.unsupported:11]: Has this test been removed?

File: tests/openssl/tests_3.unsupported [hunk 2]
```diff
@@ -38,8 +36,6 @@
 ../../3rdparty/openssl/openssl/test/siphash_internal_test.c
 # SM2 is disabled by OE.
 ../../3rdparty/openssl/openssl/test/sm2_internal_test.c
-# SM3 is disabled by OE.
-../../3rdparty/openssl/openssl/test/sm3_internal_test.c
```
Comment (mingweishih) [inline tests/openssl/tests_3.unsupported:42]: Same question for this test.

Comment (dcarpente) [reply tests/openssl/tests_3.unsupported:42]: I made a mistake in a previous PR adding this to unsupported, it's already in the supported test list

File: tests/openssl/tests_3.unsupported (hunk 1 above)
Comment (dcarpente) [reply tests/openssl/tests_3.unsupported:11]: Thank you for the catch, I think this change might be a mistake, I'll double check

Comment (dcarpente) [reply tests/openssl/tests_3.unsupported:11]: Also fixing issue where these paths refer to OpenSSL 1 repo instead of OpenSSL 3.0

**Comment by dcarpente** [comment]
bors try

**Review by radhikaj — APPROVED** [review]

**Review by mingweishih — APPROVED** [review]

**Pull request merged**